BootloaderCorePkg/Library/SmbiosInitLib/SmbiosInitLib.c

```diff
@@ -0,0 +1,389 @@
+/** @file
+
+  Copyright (c) 2017 - 2019, Intel Corporation. All rights reserved.<BR>
+  This program and the accompanying materials
+  are licensed and made available under the terms and conditions of the BSD License
+  which accompanies this distribution.  The full text of the license may be found at
+  http://opensource.org/licenses/bsd-license.php.
+
+  THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+  WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+
+**/
+
+#include <IndustryStandard/SmBios.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/DebugLib.h>
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/BootloaderCoreLib.h>
+#include <Library/SmbiosInitLib.h>
+#include <Library/PcdLib.h>
+#include <Library/BootloaderCommonLib.h>
+#include "SmbiosTables.h"
+
+VOID    *mType127Ptr      = NULL;
+
+
+/**
+  Check if the Smbios types (including the entry point struct)
+  have crossed the statically allocated size for Smbios init
+
+**/
+EFI_STATUS
+EFIAPI
+CheckSmbiosOverflow (
+  VOID
+  )
+{
+  SMBIOS_TABLE_ENTRY_POINT      *SmbiosEntry;
+
+  SmbiosEntry = (SMBIOS_TABLE_ENTRY_POINT *) PcdGet32 (PcdSmbiosTablesBase);
```

Comment:
Should we check if SmBiosEntry is NULL ?

Reply:
Added the check.